Auto update failing for msedit

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the [msedit](https://github.com/termux/termux-packages/blob/fb20a40f1b5ac556110d40398f9c0a43fe5c108d/packages/msedit) package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="console">
INFO: Updating msedit [Current version: 1.2.1]
INFO: package being updated to 2.0.0.
Downloading https://github.com/microsoft/edit/archive/refs/tags/v2.0.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100  569k  100  569k    0     0  1528k      0 --:--:-- --:--:-- --:--:-- 1528k

You are about to commit these changes:

--------------------
diff --git a/packages/msedit/build.sh b/packages/msedit/build.sh
index 401c14a72b..b533f60a3b 100644
--- a/packages/msedit/build.sh
+++ b/packages/msedit/build.sh
@@ -2,10 +2,9 @@ TERMUX_PKG_HOMEPAGE="https://github.com/microsoft/edit"
 TERMUX_PKG_DESCRIPTION="A simple editor for simple needs (Microsoft Edit)"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="1.2.1"
-TERMUX_PKG_REVISION=1
+TERMUX_PKG_VERSION="2.0.0"
 TERMUX_PKG_SRCURL="https://github.com/microsoft/edit/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz"
-TERMUX_PKG_SHA256=41c719b08212fa4ab6e434a53242b2718ba313e8d24d090f244bb857d6a9d0fd
+TERMUX_PKG_SHA256=f35da309c5f3d92b10e5c4b2267e4d5e29d809b2aa460e80326b11f7feba72a5
 TERMUX_PKG_DEPENDS="libicu"
 TERMUX_PKG_BUILD_IN_SRC=true
 TERMUX_PKG_AUTO_UPDATE=true
--------------------

bump(main/msedit): 2.0.0


Not committing to Git!

INFO: Trying to build package.
INFO: Package msedit exists in main repo.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
Loading restricted AppArmor profile...
termux - building msedit for arch aarch64...
::group::INFO: Fetching repo metadata
[packages-cf.termux.dev-apt-termux-main]: Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
[packages-cf.termux.dev-apt-termux-main]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-main]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-root]: Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
[packages-cf.termux.dev-apt-termux-root]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-root]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-x11]: Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
[packages-cf.termux.dev-apt-termux-x11]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-x11]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-main]: 
[packages-cf.termux.dev-apt-termux-main]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-main]: 100 13134  100 13134    0     0  23388      0 --:--:-- --:--:-- --:--:-- 23411
[packages-cf.termux.dev-apt-termux-main]: Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
[packages-cf.termux.dev-apt-termux-main]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-main]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-root]: 
[packages-cf.termux.dev-apt-termux-root]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-root]: 100 13320  100 13320    0     0  24887      0 --:--:-- --:--:-- --:--:-- 24897
[packages-cf.termux.dev-apt-termux-root]: Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
[packages-cf.termux.dev-apt-termux-root]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-root]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-x11]: 
[packages-cf.termux.dev-apt-termux-x11]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-x11]: 100 13120  100 13120    0     0  23649      0 --:--:-- --:--:-- --:--:-- 23682
[packages-cf.termux.dev-apt-termux-x11]: Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
[packages-cf.termux.dev-apt-termux-x11]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-x11]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-x11]: 
[packages-cf.termux.dev-apt-termux-x11]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-x11]: 100   833  100   833    0     0   4939      0 --:--:-- --:--:-- --:--:--  4958
[packages-cf.termux.dev-apt-termux-x11]: gpg: Signature made Tue 28 Apr 2026 01:58:02 PM UTC
[packages-cf.termux.dev-apt-termux-x11]: gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
[packages-cf.termux.dev-apt-termux-x11]: gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
[packages-cf.termux.dev-apt-termux-x11]: gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
[packages-cf.termux.dev-apt-termux-main]: 
[packages-cf.termux.dev-apt-termux-main]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-main]: 100   833  100   833    0     0   1535      0 --:--:-- --:--:-- --:--:--  1534
[packages-cf.termux.dev-apt-termux-main]: 100   833  100   833    0     0   1535      0 --:--:-- --:--:-- --:--:--  1534
[packages-cf.termux.dev-apt-termux-main]: gpg: Signature made Tue 28 Apr 2026 07:16:39 PM UTC
[packages-cf.termux.dev-apt-termux-main]: gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
[packages-cf.termux.dev-apt-termux-main]: gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
[packages-cf.termux.dev-apt-termux-main]: gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
[packages-cf.termux.dev-apt-termux-root]: 
[packages-cf.termux.dev-apt-termux-root]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-root]: 100   833  100   833    0     0   1503      0 --:--:-- --:--:-- --:--:--  1500
[packages-cf.termux.dev-apt-termux-root]: 100   833  100   833    0     0   1502      0 --:--:-- --:--:-- --:--:--  1500
[packages-cf.termux.dev-apt-termux-root]: gpg: Signature made Mon 27 Apr 2026 07:19:41 PM UTC
[packages-cf.termux.dev-apt-termux-root]: gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
[packages-cf.termux.dev-apt-termux-root]: gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
[packages-cf.termux.dev-apt-termux-root]: gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
::endgroup::
Downloading dependency libc++@29 if necessary...
Found libc++ in https://packages-cf.termux.dev/apt/termux-main/dists/stable
extracting libc++ to /home/builder/.termux-build/_cache-aarch64...
Downloading dependency libicu@78.3 if necessary...
Found libicu in https://packages-cf.termux.dev/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.dev/apt/termux-main/pool/main/libi/libicu/libicu_78.3_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 9971k  100 9971k    0     0  99.2M      0 --:--:-- --:--:-- --:--:-- 99.3M
extracting libicu to /home/builder/.termux-build/_cache-aarch64...
Downloading dependency icu-devtools@78.3 if necessary...
Found icu-devtools in https://packages-cf.termux.dev/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.dev/apt/termux-main/pool/main/i/icu-devtools/icu-devtools_78.3_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  205k  100  205k    0     0  3670k      0 --:--:-- --:--:-- --:--:-- 3737k
extracting icu-devtools to /home/builder/.termux-build/_cache-aarch64...
Downloading msedit@2.0.0 source from 'https://github.com/microsoft/edit/archive/refs/tags/v2.0.0.tar.gz' if necessary...
Downloading https://github.com/microsoft/edit/archive/refs/tags/v2.0.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100  569k  100  569k    0     0  1393k      0 --:--:-- --:--:-- --:--:-- 1393k
Applying patch: fix-32bit-datatypes.patch
The text leading up to this was:
--------------------------
|32-bit Android's st_dev and st_ino are 64-bit unsigned integers, not 32-bit unsigned integers
|
|https://android.googlesource.com/platform/bionic/+/refs/heads/android15-release/libc/include/sys/stat.h#83
|
|Fixes:
|error[E0308]: mismatched types
|   --> src/sys/unix.rs:380:29
|    |
|380 |         Ok(FileId { st_dev: stat.st_dev, st_ino: stat.st_ino })
|    |                             ^^^^^^^^^^^ expected `u32`, found `u64`
|
|error[E0308]: mismatched types
|   --> src/sys/unix.rs:380:50
|    |
|380 |         Ok(FileId { st_dev: stat.st_dev, st_ino: stat.st_ino })
|    |                                                  ^^^^^^^^^^^ expected `u32`, found `u64`
|
|For more information about this error, try `rustc --explain E0308`.
|error: could not compile `edit` (lib) due to 2 previous errors
|
|--- a/src/sys/unix.rs
|+++ b/src/sys/unix.rs
--------------------------
File to patch: 
Skip this patch? [y] 
1 out of 1 hunk ignored
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occurred when I last tried to update at 2026-04-28 21:48:58 UTC.<br>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/25070782888">25070782888</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

--- a/packages/msedit/build.sh
+++ b/packages/msedit/build.sh
@@ -2,17 +2,14 @@ TERMUX_PKG_HOMEPAGE="https://github.com/microsoft/edit"
 TERMUX_PKG_DESCRIPTION="A simple editor for simple needs (Microsoft Edit)"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="1.2.1"
-TERMUX_PKG_REVISION=1
+TERMUX_PKG_VERSION="2.0.0"
 TERMUX_PKG_SRCURL="https://github.com/microsoft/edit/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz"
-TERMUX_PKG_SHA256=41c719b08212fa4ab6e434a53242b2718ba313e8d24d090f244bb857d6a9d0fd
+TERMUX_PKG_SHA256=f35da309c5f3d92b10e5c4b2267e4d5e29d809b2aa460e80326b11f7feba72a5
 TERMUX_PKG_DEPENDS="libicu"
 TERMUX_PKG_BUILD_IN_SRC=true
 TERMUX_PKG_AUTO_UPDATE=true
 
 termux_step_pre_configure() {
-	# Do not pin the compiler version to nightly.
-	rm "$TERMUX_PKG_SRCDIR/rust-toolchain.toml"
 	termux_setup_rust
 }
 

--- a/packages/msedit/fix-32bit-datatypes.patch
+++ b/packages/msedit/fix-32bit-datatypes.patch
@@ -18,9 +18,9 @@ error[E0308]: mismatched types
 For more information about this error, try `rustc --explain E0308`.
 error: could not compile `edit` (lib) due to 2 previous errors
 
---- a/src/sys/unix.rs
-+++ b/src/sys/unix.rs
-@@ -360,6 +360,14 @@ pub fn open_stdin_if_redirected() -> Option<File> {
+--- a/crates/edit/src/sys/unix.rs
++++ b/crates/edit/src/sys/unix.rs
+@@ -340,6 +340,14 @@ fn set_tty_nonblocking(nonblock: bool) {
      }
  }
  

